### ISSUE TITLE
[Bug]: (-5:Bad argument) CAP_IMAGES: error in Youtube Mode

### ISSUE THREAD

**Issue by abhiTronix** [abhiTronix/vidgear#125]
## Description

Youtube have switched to **AV1** (AOMedia Video 1) format for its video streams (following Metadata is extracted with FFmpeg):
```sh
  Metadata:
    major_brand     : dash
    minor_version   : 0
    compatible_brands: iso6av01mp41
    creation_time   : 2020-04-22T02:54:59.000000Z
  Duration: 00:03:32.04, start: 0.000000, bitrate: 1817 kb/s
    Stream #0:0(und): Video: av1 (av01 / 0x31307661), none(tv, bt709), 1920x1080, 28 kb/s, 25 fps, 25 tbr, 12800 tbn, 12800 tbc (default)
    Metadata:
      creation_time   : 2020-04-22T02:54:59.000000Z
      handler_name    : ISO Media file produced by Google Inc.
```

which is [**NOT supported** ](https://github.com/opencv/opencv/issues/11389) by [OpenCV's VideoCapture API videoio backends](https://docs.opencv.org/master/d4/d15/group__videoio__flags__base.html#ga023786be1ee68a9105bf2e48c700294d) yet, and hence streaming YouTube video is not possible on vidgear too.

### Acknowledgment

- [x] A brief but descriptive Title of your issue
- [x] I have searched the [issues](https://github.com/abhiTronix/vidgear/issues) for my issue and found nothing related or helpful.
- [x] I have read the [FAQ](https://github.com/abhiTronix/vidgear/wiki/FAQ-&-Troubleshooting).
- [x] I have read the [Wiki](https://github.com/abhiTronix/vidgear/wiki#vidgear).
- [x] I have read the [Contributing Guidelines](https://github.com/abhiTronix/vidgear/blob/master/contributing.md).

### Environment

* VidGear version: 0.1.7-dev
* Branch:  Testing
* Python version: all
* pip version: all
* Operating System and version: all

### Expected Behavior

VidGear and OpenCV must be able to play pafy video streams. 

### Actual Behavior

VidGear and OpenCV fails to play pafy video streams and exit with error. 

### Try this code with VidGear and OpenCV:
```python
# import required libraries
from vidgear.gears import CamGear
import cv2


# Add YouTube Video URL as input source of your choice
stream = CamGear(source='https://youtu.be/dQw4w9WgXcQ', y_tube=True, logging=True).start() 

# loop over
while True:

    # read frames from stream
    frame = stream.read()

    # check for frame if Nonetype
    if frame is None:
        break


    # {do something with the frame here}


    # Show output window
    cv2.imshow("Output Frame", frame)

    # check for 'q' key if pressed
    key = cv2.waitKey(1) & 0xFF
    if key == ord("q"):
        break

# close output window
cv2.destroyAllWindows()

# safely close video stream
stream.stop()
```

and it will exit with this error:

```sh
CamGear :: DEBUG :: YouTube source ID: `dQw4w9WgXcQ`, Title: `Rick Astley - Never Gonna Give You Up (Video)` & Video_Extension: `mp4`
CamGear :: DEBUG :: Enabling Threaded Queue Mode for the current video source!

[ERROR:0] global /home/abhishek/Downloads/opencv-master/modules/videoio/src/cap.cpp (142) open VIDEOIO(CV_IMAGES): raised OpenCV exception:

OpenCV(4.3.0-pre) /home/abhishek/Downloads/opencv-master/modules/videoio/src/cap_images.cpp:235: error: (-5:Bad argument) CAP_IMAGES: error, expected '0?[1-9][du]' pattern, got: https://r2---sn-bpb5oupj-qxae.googlevideo.com/videoplayback?expire=1587650550&ei=lkuhXufwMcCG3LUPlbS_sAc&ip=103.206.248.130&id=o-AGlCW6TY_pX6IUWtx4Vh7I2BAtJAq2BO7_fO1uhIdSWp&itag=399&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278%2C394%2C395%2C396%2C397%2C398%2C399&source=youtube&requiressl=yes&mh=7c&mm=31%2C29&mn=sn-bpb5oupj-qxae%2Csn-qxa7sn7s&ms=au%2Crdu&mv=m&mvi=1&pcm2cms=yes&pl=24&gcr=in&initcwndbps=572500&vprv=1&mime=video%2Fmp4&gir=yes&clen=48167591&dur=212.040&lmt=1587529896388952&mt=1587628780&fvip=4&keepalive=yes&c=WEB&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=ALrAebAwRQIgDwRay_ZCsX-wt5c1EleYLdauXLnR8qItQ8QPxnDaQJoCIQDSWBilURb9HErvEGjNd40xQxwW0VQUxDWzV2VdQbdJgA%3D%3D&sig=AJpPlLswRgIhANas5JkqYMPWs7A7vckyhTYL5EPJ1Q5hmqSbBIa6BAbvAiEA3mLcmm8XhE7W18zZTTn-uf1BK9dBg19Ut8zSBkaGzys=&ratebypass=yes in function 'icvExtractPattern'


Traceback (most recent call last):
  File "t.py", line 7, in <module>
    stream = CamGear(source='https://youtu.be/dQw4w9WgXcQ', y_tube=True, logging=True).start() 
  File "/usr/local/lib/python3.6/dist-packages/vidgear/gears/camgear.py", line 235, in __init__
    "[CamGear:ERROR] :: Source is invalid, CamGear failed to intitialize stream on this source!"
RuntimeError: [CamGear:ERROR] :: Source is invalid, CamGear failed to intitialize stream on this source!

```
### 2. For confirmation. Try this Code with Pafy and OpenCV:
```python
import cv2
import pafy

video = pafy.new("dQw4w9WgXcQ")
object = video.getbestvideo()
url = object.url
print(url) #means pafy and youtube-dl working correctly

stream = cv2.VideoCapture(url)

# infinite loop
while True:
	(grabbed, frame) = stream.read()
	# read frames

	# check if frame empty
	if not grabbed:
		print('opencv is buggy')
		break

	cv2.imshow("Output Frame", frame)
	# Show output window

	key = cv2.waitKey(1) & 0xFF
	# check for 'q' key-press
	if key == ord("q"):
		#if 'q' key-pressed break out
		break

cv2.destroyAllWindows()
# close output window

stream.release()
# safely close video stream.
```

and it will exit with same error as follows:

```sh
https://r2---sn-bpb5oupj-qxae.googlevideo.com/videoplayback?expire=1587650550&ei=lkuhXufwMcCG3LUPlbS_sAc&ip=103.206.248.130&id=o-AGlCW6TY_pX6IUWtx4Vh7I2BAtJAq2BO7_fO1uhIdSWp&itag=399&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278%2C394%2C395%2C396%2C397%2C398%2C399&source=youtube&requiressl=yes&mh=7c&mm=31%2C29&mn=sn-bpb5oupj-qxae%2Csn-qxa7sn7s&ms=au%2Crdu&mv=m&mvi=1&pcm2cms=yes&pl=24&gcr=in&initcwndbps=572500&vprv=1&mime=video%2Fmp4&gir=yes&clen=48167591&dur=212.040&lmt=1587529896388952&mt=1587628780&fvip=4&keepalive=yes&c=WEB&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=ALrAebAwRQIgDwRay_ZCsX-wt5c1EleYLdauXLnR8qItQ8QPxnDaQJoCIQDSWBilURb9HErvEGjNd40xQxwW0VQUxDWzV2VdQbdJgA%3D%3D&sig=AJpPlLswRgIhANas5JkqYMPWs7A7vckyhTYL5EPJ1Q5hmqSbBIa6BAbvAiEA3mLcmm8XhE7W18zZTTn-uf1BK9dBg19Ut8zSBkaGzys=&ratebypass=yes

[ERROR:0] global /home/abhishek/Downloads/opencv-master/modules/videoio/src/cap.cpp (142) open VIDEOIO(CV_IMAGES): raised OpenCV exception:

OpenCV(4.3.0-pre) /home/abhishek/Downloads/opencv-master/modules/videoio/src/cap_images.cpp:235: error: (-5:Bad argument) CAP_IMAGES: error, expected '0?[1-9][du]' pattern, got: https://r2---sn-bpb5oupj-qxae.googlevideo.com/videoplayback?expire=1587650550&ei=lkuhXufwMcCG3LUPlbS_sAc&ip=103.206.248.130&id=o-AGlCW6TY_pX6IUWtx4Vh7I2BAtJAq2BO7_fO1uhIdSWp&itag=399&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278%2C394%2C395%2C396%2C397%2C398%2C399&source=youtube&requiressl=yes&mh=7c&mm=31%2C29&mn=sn-bpb5oupj-qxae%2Csn-qxa7sn7s&ms=au%2Crdu&mv=m&mvi=1&pcm2cms=yes&pl=24&gcr=in&initcwndbps=572500&vprv=1&mime=video%2Fmp4&gir=yes&clen=48167591&dur=212.040&lmt=1587529896388952&mt=1587628780&fvip=4&keepalive=yes&c=WEB&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=ALrAebAwRQIgDwRay_ZCsX-wt5c1EleYLdauXLnR8qItQ8QPxnDaQJoCIQDSWBilURb9HErvEGjNd40xQxwW0VQUxDWzV2VdQbdJgA%3D%3D&sig=AJpPlLswRgIhANas5JkqYMPWs7A7vckyhTYL5EPJ1Q5hmqSbBIa6BAbvAiEA3mLcmm8XhE7W18zZTTn-uf1BK9dBg19Ut8zSBkaGzys=&ratebypass=yes in function 'icvExtractPattern'


opencv is buggy

```

**Comment by abhiTronix** [comment]
**Successfully resolved and merged in commit 9fd1fc3**

**Comment by varneyraney** [comment]
Hi, 
Although camgear is working for me on the youtube video above.  I am getting the "Source is invalid error" on some youtube videos such as, https://www.youtube.com/watch?v=moFOtNAujk8, and https://www.youtube.com/watch?v=w_d4CC5gCQU. I would like to know if anyone else is experience this issue. 

I am using 
vidgear version 0.1.7-pypi
python version 3.8.3. 
pip version 19.2.3 
Windows 10
Thanks for your help, 
Varne

**Comment by abhiTronix** [comment]
@varneyraney **Are sure it is not working, because both URL are working just fine on my system.** Did you tried updating `youtube-dl`, which you can do as follows:
```sh
pip install -U youtube-dl
```

also did you used this exact code as follows:
```sh

# import required libraries
from vidgear.gears import CamGear
import cv2


# Add YouTube Video URL as input source of your choice
stream = CamGear(source='https://www.youtube.com/watch?v=w_d4CC5gCQU', y_tube=True, logging=True).start() 

# loop over
while True:

    # read frames from stream
    frame = stream.read()

    # check for frame if Nonetype
    if frame is None:
        break


    # {do something with the frame here}


    # Show output window
    cv2.imshow("Output Frame", frame)

    # check for 'q' key if pressed
    key = cv2.waitKey(1) & 0xFF
    if key == ord("q"):
        break

# close output window
cv2.destroyAllWindows()

# safely close video stream
stream.stop()
# safely close video stream.
```

**Can you share the terminal output?**

**Comment by varneyraney** [comment]
@abhiTronix thank you for replying back. 
the error is from my machine. I was able to successfully run the code you pasted above on a virtual machine(Ubuntu 18.04). But it is still not running on my computer. 
python -c "import vidgear; print(vidgear.__version__)"  gives me 0.1.7
I ran pip install -U youtube_dl
Maybe the fact that I have both python 3.5.3 and python 3.8.3 on my computer is the cause my issue. I uninstalled vidgear 0.1.6 before installing 0.1.7.

**Comment by abhiTronix** [comment]
@varneyraney Thanks for finding this bug. **I've fixed this in the latest commit 4d8814c on testing branch. Kindly reinstall latest vidgear as follows:**
```sh
  git clone https://github.com/abhiTronix/vidgear.git
  cd vidgear
  git checkout testing
  python -m pip install .
``` 

**and both youtube URLs must work on Windows too. Goodluck!**
